src/Moryx.Model/DbContextManager.cs

```diff
@@ -19,35 +20,12 @@ namespace Moryx.Model
     /// Kernel component handling data models and their runtime configurators
     /// </summary>
     [InitializableKernelComponent(typeof(IDbContextManager))]
-    public class DbContextManager : IDbContextManager, IInitializable, ILoggingHost
+    public class DbContextManager : IDbContextManager
     {
-        #region Dependencies
-
-        /// <summary>
-        /// Config manager to handle database configurations
-        /// </summary>
-        public IConfigManager ConfigManager { get; set; }
-
-        /// <summary>
-        /// Logger for the database configurators
-        /// </summary>
-        public IModuleLogger Logger { get; set; }
-
-        /// <summary>
-        /// Logger root for this component
-        /// </summary>
-        public ILoggerManagement LoggerManagement { get; set; }
-
-        #endregion
-
-        string ILoggingHost.Name => nameof(DbContextManager);
         private ModelWrapper[] _knownModels;
```

Comment:
Could be an enumerable for memory purposes, if it doesn't change many things!

Reply:
How does using an enumerable change memory footprint? Array is AFAIK the most efficient collection type and implementation of IEnumerable.

Reply:
IEnumerable is an unallocated datastructure. A good practice says that what comes from db, just pass it forward to the front end and leave it unallocated. So, as long as we don't say .ToArray or .ToList, etc. The IEnumerable is unallocated and saves memory. I am not sure if this theory is applicable in this case or not

Reply:
`IEnumerable` is just an interface, it depends on the concrete type underneath. Declaring the field as `IEnumerable` and assigning an array has the same result.

What you describe only applies to implementations of IEnumerable that only allocate when the `Enumerator` is created, which is often the cases for database queries. In that case it is recommended to pass IEnumerable to avoid allocation until it is needed. `_knownModels` on the other hand is the result of iterating the DI container and in that case storing a handful of references is more efficient that reiterating the container for every request.